docs/source/reference-core.rst

```diff
@@ -831,34 +814,33 @@ For example, here's a function that takes a list of functions, runs
 them all concurrently, and returns the result from the one that
 finishes first::
 
-   # XX this example can be simplified a little after #136 is fixed in 0.3.0
-
    async def race(*async_fns):
        if not async_fns:
            raise ValueError("must pass at least one argument")
 
-       async def racecar(results, async_fn, cancel_scope):
-           result = await async_fn()
-           results.append(result)
-           cancel_scope.cancel()
+       q = trio.Queue(1)
+
+       async def jockey(async_fn):
+           await q.put(await async_fn())
 
        async with trio.open_nursery() as nursery:
-           results = []
-           cancel_scope = nursery.cancel_scope
            for async_fn in async_fns:
-               nursery.start_soon(racecar, results, async_fn, cancel_scope)
-
-       return results[0]
-
-This works by starting a set of racecar tasks which each try to run
-their function, report back, and then cancel all the rest. Eventually
-one suceeds, all the tasks are cancelled and exit, and then our
-nursery exits and we return the winning value. And if one or more of
-them raises an unhandled exception then Trio's normal handling kicks
-in: it cancels the others and then propagates the exception. If you
-wanted different behavior, you could do that by adding a ``try`` block
-to the ``racecar`` function to catch exceptions and handle them
-however you like.
+               nursery.start_soon(jockey, async_fn)
+               winner = await q.get()
```

Comment:
IIUC this should run after the loop, not inside it.

Reply:
Nice catch! Fixed.